src/k8s/pkg/k8sd/features/cilium/internal.go

```diff
@@ -1,18 +1,78 @@
 package cilium
 
 import (
+	"fmt"
+	"slices"
+	"strconv"
+	"strings"
+
 	apiv1_annotations "github.com/canonical/k8s-snap-api/api/v1/annotations/cilium"
 	"github.com/canonical/k8s/pkg/k8sd/types"
 )
 
+const (
+	// maxVLANTags is the maximum number of VLAN tags that can be configured
+	maxVLANTags = 5
+	// minVLANIDValue is the minimum valid 802.1Q VLAN ID value
+	minVLANIDValue = 1
+	// maxVLANIDValue is the maximum valid 802.1Q VLAN ID value
+	maxVLANIDValue = 4094
+)
+
 type config struct {
 	devices             string
 	directRoutingDevice string
+	vlanBPFBypass       []int
+}
+
+func validateVLANBPFBypass(vlanList string) ([]int, error) {
+	vlanList = strings.TrimSpace(vlanList)
+	// Maintain compatibility with the Cilium chart definition
+	vlanList = strings.Trim(vlanList, "{}")
+	vlans := strings.Split(vlanList, ",")
+
+	// Special case: wildcard "0" allows all VLANs
+	if len(vlans) == 1 && strings.TrimSpace(vlans[0]) == "0" {
+		return []int{0}, nil
+	}
+
+	if len(vlans) > maxVLANTags {
```

Comment:
Why do we have this limitation? Is this imposed by cilium?

Reply:
Yes. Unfortunately, this a limitation imposed by the `cilium-agent`: https://docs.cilium.io/en/stable/configuration/vlan-802.1q/